### DISCUSSION TITLE
Extension and static folder

### DISCUSSION THREAD

**Discussion by josejachuf** [emmett-framework/emmett#284] [Q&A] [answered]
I'm trying to create an extension. I have many doubts, but for now I consult the following:
My extension I want to be an app.module. How do I tell it that my static directory is one inside the extension's project?
I can pass the template directory with template_folder or template_path, but to static I don't know how to do it
What I want is for everything to be self-contained in the project directory
Jose

**Comment by gi0baro** [comment, marked as answer]
Hi @josejachuf, it's not possible to have a separated static source, thus you have 2 options:

- copying your static files to the application `static` folder in your extension `on_load` method
- add a route within your module to serve those static files from the correct location

The 2nd option would be something like this:

```python
from emmett import response
from emmett.http import HTTPFile

def stream_route():
    full_path = "something/in/your/module.js"
    raise HTTPFile(
        full_path,
        headers=response.headers,
        cookies=response.cookies
    )
```

then in your `on_load` bind the route:

```python
def on_load(self):
    self.app.route("/someurl")(stream_route)
```

you can probably get the paths dynamically with something like `os.path.dirname(os.path.abspath(__file__))`.
